error-codes/README.md

```diff
@@ -0,0 +1,97 @@
+QUIC's [STOP_SENDING](https://www.rfc-editor.org/rfc/rfc9000.html#section-3.5-4)
+frame that would tell the peer that the node has stopped reading. So there's no
+way of signaling an error while closing the read end of the stream on a yamux
+connection.
+
+### Connection Close and Stream Reset Error Codes
+Error codes are defined separately for Connection Close and Stream Reset. Stream
+Reset errors are from the range 0 to 5000 and Connection Close errors are from
+the range 5001 to 10000. Having separate errors for Connection Close and stream
+reset requires some overlap between the error code definitions but provides more
+information to the receiver. Receiving a `Bad Request: Connection Closed` error
+on reading from a stream also tells the receiver that no more streams can be
+started on the same connection. Implementations MUST provide the Connection
+Close error code on streams that are reset as a result of remote closing the
+connection. 
+
+For stream resets, when the underlying transport supports it, implementations
+SHOULD allow sending an error code on both closing the read side of the stream
+and resetting the write side of the stream. 
+
+## Libp2p Error Codes
+TODO!
+
+## Wire Encoding
+Different transports will encode the 32bit error code differently. 
+        
+### QUIC
+QUIC provides the ability to send an error on closing the read end of the
+stream, reseting the write end of the stream and on closing the connection. 
+
+For stream resets, the error code MUST be sent on the `RESET_STREAM` or the
+`STOP_SENDING` frame using the `Application Protocol Error Code` field as per
+the frame definitions in the
+[RFC](https://www.rfc-editor.org/rfc/rfc9000.html#name-reset_stream-frames).
+
+For Connection Close, the error code MUST be sent on the CONNECTION_CLOSE frame
+using the Error Code field as defined in the
+[RFC](https://www.rfc-editor.org/rfc/rfc9000.html#section-19.19-6.2.1).
+
+### Yamux
+Yamux has no `STOP_SENDING` frame, so there's no way to signal an error on
+closing the read side of the stream.
+
+For Connection Close, the 32bit Length field is to be interpreted as the error
+code. The error code MUST be sent as an  Big Endian unsigned 32 bit integer. 
+
+For Stream Resets, the error code is sent as the first 4 bytes of the Data Frame
+following the header with RST flag set as defined in the [yamux spec
+extension](https://github.com/libp2p/specs/pull/622).
+
+Note: On TCP connections with `SO_LINGER` set to 0, the error code on connection
+close may not be delivered.  
+
+### WebRTC
+Since WebRTC doesn't provide reliable delivery for frames that are followed by
+closing of the underlying data channel, there's no simple way to introduce error
+codes in the current implementation. Consider the most common use case of
+resetting both read and write sides of the stream and closing the data channel.
+The chrome implementation will not deliver the final `RESET_STREAM` msg and
```

Comment:
Is this a reference to [this Chrome bug](https://issues.chromium.org/issues/40072842)?

It's been fixed so Chrome's datachannel message delivery should be reliable now.

Reply:
Yes it is. I didn't know chrome fixed it. We can introduce error codes in webrtc then.